src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/SecretsMasker.java

```diff
@@ -71,19 +72,28 @@ public OutputStream decorate(OutputStream logger) throws IOException, Interrupte
             protected void eol(byte[] b, int len) throws IOException {
                 String s = new String(b, 0, len, StandardCharsets.UTF_8);
                 for (String value : values) {
-                    s = s.replace(value, "********");
+                    s = mask(s, value);
                 }
                 logger.write(s.getBytes(StandardCharsets.UTF_8));
             }
+
             @Override
             public void flush() throws IOException {
                 logger.flush();
             }
+
             @Override
             public void close() throws IOException {
                 super.close();
                 logger.close();
             }
+
+            private String mask(String s, String value) {
+                if (value != null && value.length() > 0) {
```

Comment:
```suggestion
                if (!value.isEmpty()) {
```

Reply:
@jglick out of curiosity, are we sure we're not putting in any `null` values there? I couldn't find and promise like that, hence the null check.

Reply:
Manual analysis says no. It would be nice to `assert` that in the `SecretsMasker` constructor for clarity.